diff --git a/install.sh b/install.sh
index 9d46dc3c..d793afd2 100755
--- a/install.sh
+++ b/install.sh
@@ -109,8 +109,6 @@ install_theme() {
   ln -sr "${THEME_DIR}/24"                                                       "${THEME_DIR}/24@2x"
   ln -sr "${THEME_DIR}/32"                                                       "${THEME_DIR}/32@2x"
   ln -sr "${THEME_DIR}/scalable"                                                 "${THEME_DIR}/scalable@2x"
-
-  gtk-update-icon-cache "${THEME_DIR}"
 }
 
 while [ $# -gt 0 ]; do
