--- a/zfs-0.7.9/module/zfs/metaslab.c	2018-05-08 22:45:12.000000000 +0200
+++ b/zfs-0.7.9/module/zfs/metaslab.c	2018-10-16 15:00:55.337358458 +0200
@@ -1049,11 +1049,11 @@
 }
 
 static range_tree_ops_t metaslab_rt_ops = {
-	metaslab_rt_create,
-	metaslab_rt_destroy,
-	metaslab_rt_add,
-	metaslab_rt_remove,
-	metaslab_rt_vacate
+  .rtop_create = metaslab_rt_create,
+	.rtop_destroy = metaslab_rt_destroy,
+	.rtop_add = metaslab_rt_add,
+	.rtop_remove = metaslab_rt_remove,
+	.rtop_vacate = metaslab_rt_vacate
 };
 
 /*
